apis/k8ssandra/v1alpha1/k8ssandracluster_types.go

```diff
@@ -263,8 +268,10 @@ type CassandraDatacenterTemplate struct {
 
 // DatacenterOptions are configuration settings that are can be set at the Cluster level and overridden for a single DC
 type DatacenterOptions struct {
-	// ServerVersion is the Cassandra version.
-	// +kubebuilder:validation:Pattern=(3\.11\.\d+)|(4\.0\.\d+)
+	// ServerVersion is the Cassandra or DSE version. The following versions are supported:
+	// - Cassandra: 3.11.X and 4.0.X
+	// - DSE: 6.8.X
+	// +kubebuilder:validation:Pattern=(6\.8\.\d+)|(3\.11\.\d+)|(4\.0\.\d+)
```

Comment:
Do we explicitly want to add a pattern on versions that is so specific?
Updating it each time a new version comes out seems a little bit overkill, doesn't it?

Reply:
Yes I agree. I copied the pattern from cass-operator, see [here](https://github.com/k8ssandra/cass-operator/blob/master/apis/cassandra/v1beta1/cassandradatacenter_types.go#L91). But I would be in favor of leaving this field free-form or just validating a general semver string.

Reply:
Hmm, we actually use this one: 

https://github.com/k8ssandra/cass-operator/blob/master/pkg/images/images.go#L27

That kubebuilder one should be removed, I don't even know if it applies..